src/App.jsx

```diff
@@ -14,33 +18,82 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function getSortedGoods(goods, sortField) {
+  const sortedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  switch (sortField) {
+    case 'alphabetically':
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+      break;
+    case 'byLength':
+      sortedGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      break;
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  return sortedGoods;
+}
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+const SORT_FIELDS = {
+  alphabetically: 'alphabetically',
+  byLength: 'byLength',
+  reverse: 'reverse',
+};
+
+export const App = () => {
+  const [sortedField, setSortedField] = useState('');
+  const [reversed, setReversed] = useState(false);
+  let visibleGoods = getSortedGoods(goodsFromServer, sortedField);
+
+  if (reversed) {
+    visibleGoods = visibleGoods.toReversed();
```

Comment:
The method `toReversed` is not a standard JavaScript method for arrays. You might want to use the `reverse()` method instead, which reverses the array in place. If you need a non-mutating version, consider using `slice().reverse()`.